### PR TITLE
Editing - Nodes orchestration

### PR DESCRIPTION
**Pull request by alaudazzi**
This PR:

- Adds a TOC at the top of the page to easy the navigation (this page is very long)
- Fixes some `code` typeface
- Tries to reorganize bulleted lists to facilitate the reading
- Suggests a few edits

Relates to [#63068](https://github.com/elastic/cloud/issues/63068)

### CODE REVIEW

File: docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc [hunk 1]
```diff
@@ -54,88 +62,118 @@ spec:
...
+*  `discovery.seed_hosts`
+*  `cluster.initial_master_nodes`
+*  `discovery.zen.minimum_master_nodes`
+*  `_cluster/voting_config_exclusions`
 
 [id="{p}-orchestration-limitations"]
 == Limitations
 
-Based on how Kubernetes and `StatefulSets` operate, ECK orchestration has the following limitations:
+Based on how Kubernetes and StatefulSets operate, ECK orchestration has the following limitations:
```
Comment (charith-elastic) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc:145]:
```suggestion
Due to relying on Kubernetes primitives such as StatefulSets, the ECK orchestration process has some inherent limitations.
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc [hunk 2]
```diff
@@ -54,88 +62,118 @@ spec:
...
 
-* Storage requirements (including volume size) of an existing `NodeSet` cannot be updated. link:https://github.com/kubernetes/enhancements/issues/661[StatefulSet volumes expansion is not available in Kubernetes yet]. To upgrade the storage size, you can create a new `NodeSet`, or rename an existing one. Renaming a `NodeSet` automatically creates a new `StatefulSet` with the specified storage size. The original `StatefulSet` is removed once the Elasticsearch data is migrated to the nodes of the new `StatefulSet`.
+* Storage requirements (including volume size) of an existing NodeSet cannot be updated. link:https://github.com/kubernetes/enhancements/issues/661[StatefulSet volumes expansion] is not available in Kubernetes yet. To upgrade the storage size, you can create a new NodeSet, or rename an existing one. Renaming a NodeSet automatically creates a new StatefulSet with the specified storage size. The original StatefulSet is removed once the Elasticsearch data is migrated to the nodes of the new StatefulSet.
```
Comment (charith-elastic) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc:147]:
```suggestion
* Storage requirements (including volume size) of an existing NodeSet cannot be updated. link:https://github.com/kubernetes/enhancements/issues/661[StatefulSet volumes expansion] is not available in Kubernetes yet. To upgrade the storage size, you can add a new NodeSet or modify an existing one by renaming it first. Changing the name of the NodeSet  triggers the creation of a new StatefulSet which will then inherit the newly updated storage size as well. ECK will ensure that the data from the old StatefulSet is migrated to the new one before the old one is deleted.
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc [hunk 3]
```diff
@@ -54,88 +62,118 @@ spec:
...
 
-* Cluster availability is not be guaranteed in the following cases:
+* Cluster availability is not guaranteed in the following cases:
```
Comment (charith-elastic) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc:149]:
```suggestion
* Cluster availability during rolling upgrades is not guaranteed for the following cases:
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc [hunk 4]
```diff
@@ -54,88 +62,118 @@ spec:
...
 
 ** During the rolling upgrade of single-node clusters
 
 ** For clusters that have indices with no replicas
```
Comment (charith-elastic) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc:153]:
```suggestion
** Single-node clusters
** Clusters containing indices with no replicas
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc [hunk 5]
```diff
@@ -54,88 +62,118 @@ spec:
...
+If an Elasticsearch node holds the only copy of a shard, this shard becomes unavailable while the node is upgraded. Clusters with more than one node and at least one replica per index are recommended.
 
-* Elasticsearch `Pods` may stay `Pending` during a rolling upgrade if the Kubernetes scheduler cannot re-schedule them back. This is especially important when using local `PersistentVolumes`. If the Kubernetes node bound to a local `PersistentVolume` does not have enough capacity to host an upgraded `Pod` which was temporarily removed, that `Pod` will stay Pending.
+* Elasticsearch Pods may stay `Pending` during a rolling upgrade if the Kubernetes scheduler cannot re-schedule them back. This is especially important when using local PersistentVolumes. If the Kubernetes node bound to a local PersistentVolume does not have enough capacity to host an upgraded Pod which was temporarily removed, that Pod will stay `Pending`.
 
 * Rolling upgrades can make progress if the Elasticsearch cluster health is green. ECK will also make progress if the cluster health is yellow under the following conditions:
```
Comment (charith-elastic) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc:159]:
```suggestion
* Rolling upgrades can only make progress if the Elasticsearch cluster health is green. There are exceptions to this rule if the cluster health is yellow and if the following conditions are satisfied:
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc [hunk 6]
```diff
@@ -54,88 +62,118 @@ spec:
...
-** A cluster version upgrade is in progress and some `Pods` are not up to date
-** There are no initializing or relocating shards
+** A cluster version upgrade is in progress and some Pods are not up to date.
```
Comment (charith-elastic) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc:160]:
```suggestion
** A cluster version upgrade is in progress and some Pods are not up-to-date.
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc [hunk 7]
```diff
@@ -54,88 +62,118 @@ spec:
...
+** There are no initializing or relocating shards.
 
-If the above conditions are met, then ECK can delete a `Pod` for upgrade even if the cluster health is yellow as long as the `Pod` is not holding the last available replica of a shard.
+If the above conditions are met, then ECK can delete a Pod for upgrade even if the cluster health is yellow, as long as the Pod is not holding the last available replica of a shard.
 
 The health of the cluster is deliberately ignored in the following cases:
 
-** If all the Elasticsearch nodes of a `NodeSet` are unavailable, probably caused by a misconfiguration, the operator ignores the cluster health and upgrades nodes of the `NodeSet`.
+** Is all the Elasticsearch nodes of a NodeSet are unavailable, probably caused by a misconfiguration, the operator ignores the cluster health and upgrades nodes of the NodeSet.
```
Comment (charith-elastic) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc:167]:
```suggestion
** If all the Elasticsearch nodes of a NodeSet are unavailable, probably caused by a misconfiguration, the operator ignores the cluster health and upgrades nodes of the NodeSet.
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc [hunk 8]
```diff
@@ -54,88 +62,118 @@ spec:
...
 ** If an Elasticsearch node to upgrade is not healthy, and not part of the Elasticsearch cluster, the operator ignores the cluster health and upgrades the Elasticsearch node.
```
Comment (charith-elastic) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc:168]: This line doesn't make sense to me. Is it possible to have an Elasticsearch node that doesn't belong to a cluster?

Comment (alaudazzi) [reply docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc:168]: I guess it was just a typo from my side.

File: docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc (hunk 6 above)
Comment (alaudazzi) [reply docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc:160]: Hyphenate it only when used as an adjective => an up-to-date Pod.

File: docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc [hunk 9]
```diff
@@ -54,88 +62,118 @@ spec:
...
+
+The Elasticsearch cluster is composed of 13 nodes: 3 master nodes and 10 data nodes.
 
 [id="{p}-upgrading"]
 == Upgrading the cluster
 
-ECK handles smooth upgrades from one cluster specification to another. You can apply a new Elasticsearch specification at any time.
+ECK handles smooth upgrades from one cluster specification to another. You can apply a new Elasticsearch specification at any time. For example, based on the Elasticsearch specification described in the <<{p}-nodesets,NodeSets overview>>, you can:
+
+*  Add five additional Elasticsearch data nodes: In `data-nodes` change the value in the `count` field from 10 to 15.
```
[the remark this answers is not in the document]
Comment (alaudazzi) [reply docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc:77]: It's hard to say. Might even become heavier. I would leave it as is.

File: docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc [hunk 10]
```diff
@@ -54,88 +62,118 @@ spec:
...
-* When an existing `NodeSet` is renamed, ECK performs the creation of a new `NodeSet` with the new name, and the removal of the old `NodeSet`, according to the `NodeSet` creation and removal patterns described above. Elasticsearch data is migrated away from the deprecated `NodeSet` before removal. The Elasticsearch resource <<{p}-update-strategy,update strategy>> controls how many nodes can exist above or below the target node count during the upgrade.
-
-In all these cases, ECK handles `StatefulSet` operations according to the Elasticsearch orchestration best practices, by adjusting the orchestration settings `discovery.seed_hosts`, `cluster.initial_master_nodes`, `discovery.zen.minimum_master_nodes`, and `_cluster/voting_config_exclusions` accordingly.
+Depending on how the NodeSets are updated, ECK handles the Kubernetes resources reconciliation in various ways.
+
+* A new NodeSet is added to the Elasticsearch resource.
```
[the remark this answers is not in the document]
Comment (alaudazzi) [reply docs/orchestrating-elastic-stack-applications/elasticsearch/orchestration.asciidoc:116]: Same comment above. Might be heavier.

**Review by charith-elastic — APPROVED** [review]

**Pull request merged**